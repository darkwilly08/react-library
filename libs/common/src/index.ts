export * from './lib/hooks/useDebounce';
export * from './lib/hooks/useIsFirstRender';
export * from './lib/hooks/useScreenSize';
export * from './lib/hooks/useIsSsr';
