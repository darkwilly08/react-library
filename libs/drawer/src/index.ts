export * from './lib/drawer';
export * from './lib/refDrawer';
export * from './lib/models';
