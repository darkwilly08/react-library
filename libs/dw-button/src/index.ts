export * from './lib/dw-button';
