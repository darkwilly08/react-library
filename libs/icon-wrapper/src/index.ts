export * from './lib/icon-wrapper';
