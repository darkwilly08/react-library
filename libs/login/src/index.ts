export * from './lib/login';
export * from './lib/hooks/useAuth';
export * from './lib/contexts/userContext';
