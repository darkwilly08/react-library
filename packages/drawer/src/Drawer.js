import clsx from 'clsx';

import './styles/Drawer.scss';

const Drawer = ({ name, className }) => {
  const rootClass = 'drawer';
  return <div className={clsx(rootClass, className)}>{name}</div>;
};

export { Drawer };
