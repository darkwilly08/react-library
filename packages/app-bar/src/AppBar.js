// import { IconWrapper } from '../../icon-wrapper/index.js';
import { IconWrapper } from '@darkwilly08/icon-wrapper';
import './styles/AppBar.scss';

const AppBar = () => {
  const rootClass = 'app-bar';
  return (
    <div className={rootClass}>
      <IconWrapper name="menu" />
      <div>title</div>
      <div className={`${rootClass}__right`}>test</div>
    </div>
  );
};

export { AppBar };
