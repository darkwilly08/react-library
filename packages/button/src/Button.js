import Ripples from 'react-ripples';
import clsx from 'clsx';

import { IconWrapper } from '@darkwilly08/icon-wrapper';

import './styles/Button.scss';

const Button = ({ text, leftIcon, rightIcon, flat, color, textColor, onClick }) => {
  const style = {
    color: textColor ?? null,
    backgroundColor: color ?? null,
  };
  return (
    <Ripples className={clsx(!flat && 'shadow')} onClick={onClick}>
      <button className={clsx('button', flat && 'button--flat')} style={style}>
        {leftIcon && <IconWrapper name={leftIcon} className="button__left-icon" />}
        <div className="button__text">{text}</div>
        {rightIcon && <IconWrapper name={rightIcon} className="button__right-icon" />}
      </button>
    </Ripples>
  );
};

export { Button };
