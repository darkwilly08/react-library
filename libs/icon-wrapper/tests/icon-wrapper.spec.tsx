import { render } from '@testing-library/react';

import { IconWrapper, IconWrapperProps } from '../src';

describe('IconWrapper', () => {
  const mockProps: IconWrapperProps = {
    name: 'home',
  };
  it('should render successfully', () => {
    const { container } = render(<IconWrapper name={mockProps.name} />);
    expect(container).toMatchSnapshot();
    expect(container).not.toBeEmptyDOMElement();
  });
});
