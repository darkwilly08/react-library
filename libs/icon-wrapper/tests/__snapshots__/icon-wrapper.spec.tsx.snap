// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconWrapper should render successfully 1`] = `
<div>
  <span
    class="material-icons"
  >
    home
  </span>
</div>
`;
